### PR TITLE
fix(array checking css): issue #232

### PR DESCRIPTION
**Pull request by nmccready**
Resolves issue #232

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -140,7 +145,9 @@ module.exports.init = function init(options) {
         sourceMap = generator.toJSON();
 
       } else if (fileType === '.css') {
+        debug('csss');
```
Comment (nmccready) [inline index.js]: need to fix typo

File: index.js [hunk 2]
```diff
@@ -17,6 +17,8 @@ var urlRegex = /^(https?|webpack(-[^:]+)?):\/\//;
  * Initialize source mapping chain
  */
 module.exports.init = function init(options) {
+  var debug = require('debug')('gulp-sourcemaps:init');
```
Comment (nmccready) [inline index.js]: debug logging for my sanity

File: index.js [hunk 3]
```diff
@@ -150,10 +157,13 @@ module.exports.init = function init(options) {
             });
           }
           for (var key in ast) {
+            debug('key: ' + key);
+            debug(ast[key]);
             if (key !== "position") {
               if (Object.prototype.toString.call(ast[key]) === '[object Object]') {
                 registerTokens(ast[key]);
-              } else if (ast[key].constructor === Array) {
+              } else if (Array.isArray(ast[key])) {
```
Comment (nmccready) [inline index.js]: main bug fix

**Comment by nmccready** [comment]
@shinnn @hummal @floridoo @zacronos @SamanthaAdrichem

**Comment by nmccready** [comment]
Once merged I will roll this into master

**Comment by nmccready** [comment]
I am only PR'ing this to see if anyone has any qualms with debug being added as a dependency

**Comment by shinnn** [comment]
I'm not a good fan of `debug`... in exchange for the good debug log, it makes the code a bit more messy.

======

It's just my opinion. Feel free to merge this if others vote for this PR.

**Comment by nmccready** [comment]
@shinnn curious what is exactly the beef with debug? I find it rather clean. If anything I have issues with the numerous outstanding PR's they have on the repo. Do you have better alternatives?

One thing we have done at my office is basically wrap debug to evaluate functions only if they are enabled. That way it saves on memory and string concatenation.

Example:


```
var bigHarryString = 'lots of string';
debug(function(){return 'bigHarryString :' + bigHarryString;}); //internal closure is not called if that debug namespace is not enabled.
```

**Comment by shinnn** [comment]
If you're so confident about what you do, no need to ping others. Just merge it.

**Comment by nmccready** [comment]
@shinnn yeah, I am working on a debug extension to keep the debug overhead minimal

File: test/consolerecorder.js [hunk 4]
```diff
@@ -1,21 +0,0 @@
-'use strict';
```
Comment (nmccready) [inline test/consolerecorder.js]: no longer needed using `hook-std` instead

**Pull request merged**